packages/inertia/src/router.ts

```diff
@@ -90,9 +90,14 @@ export class Router {
     }
   }
 
-  protected restoreScrollPositions(): void {
+  protected restoreScrollPositions(attempt = 1): void {
     if (this.page.scrollRegions) {
-      this.scrollRegions().forEach((region: Element, index: number) => {
+      const availableRegions = this.scrollRegions()
+      if (attempt <= 20 && (this.page.scrollRegions.length !== availableRegions.length)) {
+        setTimeout(() => this.restoreScrollPositions(attempt + 1), 2)
+        return
+      }
```

Comment:
@claudiodekker Can we use `queueMicrotask` instead?

Or even `Promise.resolve().then(() => {})`